core/server/master/src/main/java/alluxio/master/file/DefaultFileSystemMaster.java

```diff
@@ -1298,6 +1346,10 @@ public void delete(AlluxioURI path, DeleteOptions options) throws IOException,
       if (!options.isAlluxioOnly()) {
         mMountTable.checkUnderWritableMountPoint(path);
       }
+      // Possible ufs sync.
+      syncMetadata(journalContext, inodePath, lockingScheme);
+      inodePath.validatePath(lockingScheme);
```

Comment:
why do we need to add `validatePath` now? Previously, how would the code handle the path not existing?

Reply:
Previously, we would use the `lockFullInodePath()` method, which would fail if the full path did not exist. Now, we cannot lock the full path, since the sync might load the path. So, for some RPCs we will have to check if the full path exists after the sync.